File: cpp/LAB6/LAB6.cpp
#include <iostream>
#include <cmath>
using namespace std;

float recursive(int m){
    if(m==0){
        return 0;
    }
    else {
        return (pow((-1),(m+1))/m) + recursive((m-1)); // ^ is NOT the power operator. Disappointing
    }
}

float recursive(){
    int m = 0;
    cout << "Please enter an m (Overloaded!): ";
    cin >> m;

    if(m==0){
        return 0;
    }
    else {
        return (pow((-1),(m+1))/m) + recursive((m-1)); // ^ is NOT the power operator. Disappointing
    }
}

int main() {
    int x = 0;
    cout << "Please enter a number: ";
    cin >> x;
    cout << recursive(x) << endl;
    cout << recursive() << endl;
}

File: py/LAB6/LAB6.py
print("e^n Calculator v.0.0.1")
n=int(input("Please enter a number n: "))
x=int(input("Please enter a number x: "))
factorial = lambda a: a*factorial(a-1) if a > 1 else 1
term = lambda up, down: up**down/factorial(down)
terms = sum([term(n,i) for i in range (x+1)])
print(terms)


result=0
def recursive(m):
    """This function calculates(-1^(k-1)/k) until a given m from 1."""
    global result
    if m == 0:
        print (result)
    else:
        result += (((-1)**(m+1))/m)
        recursive(m-1)
m = int(input("Please enter an m:"))
recursive(m)




